geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreAttributes.java

```diff
@@ -42,14 +43,20 @@
   public long timeInterval;
 
   public int[] diskDirSizes;
-
+  private DiskDirSizesUnit diskDirSizesUnit;
```

Comment:
This class is Serializable and you are adding a field to it. That is okay (since the serialVersionUID is set) but if we try to deserialize an older version of the class this new field, diskDirSizesUnit, will get a default value. I'm not sure for enums but it is probably "null" by default. If it is null then I think we are in trouble because the gettor would just return null. If it is null then you could change the gettor to check for that and return DEFAULT_DISK_DIR_SIZES_UNIT.  You could serialize this class to a file without the enum field, add the field in, deserialize from the file with the new version of the class and see what the enum field value defaults to in that case.

Reply:
I have modified the getter to set a default value if the attribute is null. 
I did a test with a dummy class where I added an enum field and tried to deserialize an object which was serialized before the change. And the enum field was set to null.